packages/components/src/MultiSelect/subcomponents/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useRef, InputHTMLAttributes } from "react"
+import classnames from "classnames"
+import { CheckIcon } from "~icons/CheckIcon"
+import { MinusIcon } from "~icons/MinusIcon"
+import { OverrideClassName } from "~types/OverrideClassName"
+import styles from "./Checkbox.module.scss"
+
+export type CheckedStatus = "checked" | "unchecked" | "indeterminate"
+
+type ReadOnly = {
+  readOnly: true
+  onChange?: never
+}
+
+type Controlled = {
+  readOnly?: false
+  onChange: Required<InputHTMLAttributes<HTMLInputElement>["onChange"]>
+}
+
+export type CheckboxProps = {
+  checkedStatus: CheckedStatus
+  value?: string
```

Comment:
This doesn't need to typed explicitly by the looks

Reply:
Nice catch